lib/ansible/modules/source_control/git.py

```diff
@@ -977,6 +1011,46 @@ def main():
     track_submodules = module.params['track_submodules']
 
     result.update(before=None)
+
+    if archive:
+        if module.check_mode:
+            result.update(changed=True)
+            module.exit_json(**result)
+        else:
```

Comment:
You could remove the else since we won't get here regardless in the case of check mode.